app/javascript/Pages/Classes.jsx

```diff
@@ -11,8 +11,12 @@ const Classes = (props) => {
   const [categories, setCategories] = useState([]);
   const [skills, setSkills] = useState([]);
   const [classes, setClasses] = useState([]);
-  const [categoryFilter, setCategoryFilter] = useState({ id: "" });
-  const [skillFilter, setSkillFilter] = useState({});
+  const [categoryFilter, setCategoryFilter] = useState({
+    id: props.category_id ? props.category_id : "",
```

Comment:
**issue (blocking)**: you don't need to do this so verbose (repetitive). We have 2 options now: [Optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) and [Logical OR](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR)
```js
id: props?.category_id // will be "undefined" if there is no category_id
```
```js
id: props.category_id || "" // will be empty string if not defined
```

However, in this case, wouldn't it be better to use the defaultProps option?